### PR TITLE
Add support for APIv2

### PR DESCRIPTION
**Pull request by polishmatt**

### CODE REVIEW

File: src/JwplatformClient.php [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+     *
+     * @param string $method HTTP request method
+     * @param string $path Resource or endpoint to request
+     * @param array $body HTTP request body which will be JSON encoded
+     * @param array $headers Additional HTTP request headers
+     * @param array $query_params Additional query parameters to add to the URI
+     * @return mixed
+     */
+    public function request($method, $path, $body = null, $headers = [], $query_params = null) {
+        $headers[] = "User-Agent: ";
```
Comment (boyntoni) [inline src/JwplatformClient.php:77]: Should this have a default?

File: src/JwplatformClient.php [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+        curl_setopt($curl, CURLOPT_HTTPHEADER, array(
+            'Content-Type:',
+        ));
+        $response = curl_exec($curl);
+        $curl_info = curl_getinfo($curl);
+        curl_close($curl);
+
+        $header_size = $curl_info['header_size'];
+        $header = substr($response, 0, $header_size);
+        preg_match("/ETag: \"(.*?)\"\r\n/", $header, $matches);
```
Comment (boyntoni) [inline src/JwplatformClient.php:284]: Do we implement this behavior in other clients? Don't remember seeing this in the python v2 client

**Review by boyntoni — APPROVED** [review]
Just a few comments, overall seems sane and consistent with other clients to me

File: src/JwplatformClient.php (hunk 2 above)
Comment (polishmatt) [reply src/JwplatformClient.php:284]: Not every client but it's a nice to have feature that was added to the python one: https://github.com/jwplayer/jwplatform-py/blob/e9db16462daf9931ac0e9f616c58ef529eb69aa1/jwplatform/upload.py#L227

File: src/JwplatformClient.php (hunk 1 above)
Comment (polishmatt) [reply src/JwplatformClient.php:77]: Yep I meant to add one here but forgot it.

**Pull request merged**